### PR TITLE
Grant condition to units closest to the crate

### PR DESCRIPTION
**Pull request by PunkPun**
I noticed this bug when testing #20809. It feels very off to randomly select units to be iron curtained rather than by distance

### CODE REVIEW

**Comment by PunkPun** [comment]
With the merge of #20809 I'd recommend taking this PR for next release

File: OpenRA.Mods.Common/Traits/Crates/GrantExternalConditionCrateAction.cs [hunk 1]
```diff
@@ -60,38 +60,39 @@ public override int GetSelectionShares(Actor collector)
 
 		public override void Activate(Actor collector)
 		{
-			var actorsInRange = self.World.FindActorsInCircle(self.CenterPosition, info.Range)
-				.Where(a => a != self && a != collector && a.Owner == collector.Owner && AcceptsCondition(a));
+			collector.World.AddFrameEndTask(w =>
+			{
+				if (collector.IsInWorld && !collector.IsDead)
```
Comment (penev92) [inline OpenRA.Mods.Common/Traits/Crates/GrantExternalConditionCrateAction.cs]: Like the `Where()` below, this should probably also check for `AcceptCondition`.

Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/Crates/GrantExternalConditionCrateAction.cs]: As this isn't searching for valid actors it's an unnecessary check. The function already takes care of it.

**Comment by Mailaender** [comment]
It still says
> [Desc("Grants a condition on the collector.")]

**Comment by PunkPun** [comment]
Which is true. It will give the condition to the collector no matter what.

This PR is just fixing the logic for `MaxExtraCollectors`

**Review by Mailaender — APPROVED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/6c109368bfa113c01469ee590106b4d914a17af2...8ca4af6cb55988091ee59edb487f385cff638cc5)

**Comment by Mailaender** [comment]
a4821b51a2dc724363c1df0b27f95ca7b2e60611

**Pull request merged**